docs/trace/extending-the-sdk/README.md

```diff
@@ -335,6 +335,22 @@ Custom resource detectors can be implemented:
 
 A demo ResourceDetector is shown [here](./MyResourceDetector.cs).
 
+## Sampler Detector
+
+OpenTelemetry .NET SDK provides a sampler detector for detecting sampler
```

Comment:
mm.. I am not sure of the need of such a new api. Can we simply modify SDK to automatically pick Sampler from Env Variables?

Reply:
I see some benefits for having builder with detectors:
 - possibility to programmatically disable env. vars.
 - easily add more samplers supported by the env. vars. You can just implement externally support for other samplers without taking care about parsing values for common samplers. Please consider something like `XRayOTelEnvVarSamplerDetector`. With current design it could be chained with current CommonEnvVar Detector.

Reply:
There are issues with the existing environment variable stuff. Primarily, it doesn't work with IConfiguration/IOptions. This is something I am looking at, hoping to find a new pattern that smooths all of this out. Probably first to drive #3583. I would vote to wait on this sorry @Kielek I know that is probably frustrating!

Reply:
If we speaking about timeframe, it will be great to include it at least based env. var. support for 1.4.0 release. We would like to give AutoInstrumentation users some flexibility here.

I know that we will need more thing configurable in the future so I will observe issue you have mentioned.